### DISCUSSION TITLE
Já te pediram para fazer um layout?

### DISCUSSION THREAD

**Discussion by phsantiago** [frontendbr/forum#855] [Discussão]
#### Tenho quase certeza que se você trabalha como front end a resposta do título é sim!

Aceitei um freela esses dias e me deram um briefing ótimo sobre a ideia e o que imaginam que deve ser feito, mas não deram layout, apenas um wireframe bem curto, sem nenhum texto nem informações estruturadas. Coisa do tipo: quero um dashboard para cadastrar tal tipo de produto. Só isso! Parei para pensar e isso acontece muito.

Em meio a empresas onde já tem processos estruturados isso acontece bem pouco. O ideal era que não acontecesse, mas sempre vai ter aquela validação de formulário que o designer não colocou no layout e você vai ter que se ferrar para encaixar uma cor e uma mensagem mostrando o campo invalido, ou um go horse de um lightbox para alguma campanha X.

Eu geralmente tento fazer primeiro, se não der certo já apelo para mandarem para um designer. Gostaria de saber como vocês agem em meio a essa situação que com certeza já vivenciaram ou faz parte do dia a dia.

**Comment by raffaeldantass** [comment]
Cara, eu já passei por isso
Várias vezes aliás. 

Mas eu sempre gostei muito de design, então sempre que enfrento uma situação assim já começo a imaginar um layout, desenhar (em um papel mesmo) a diagramação/layout e depois começo a passar para o photoshop. 

Geralmente eu só começo a codar depois que o layout foi aprovado. 
Nos poucos freelas que eu pego, eu já faço tudo

E particularmente eu prefiro assim pq posso controlar tudo, assim controlo melhor a questão de **performance X design**

Enfrentando esse tipo de situação consegui aprender muito sobre alguns princípios de design, UX e até um pouquinho de arquitetura da informação. 
Isso me ajudou muito no meu trabalho hj já consigo ficar mais próximo do pessoal de design, evitando layouts absurdos e as vezes sem sentido.

**Comment by kvnol** [comment]
Isso nunca aconteceu comigo, infelizmente, tirando as vezes que eu pedi. Eu gosto muito de desenhar o layout e também sempre começo pelo papel, passo para o Photoshop/Sketch/etc e depois começo a programar.

Isso é bom por várias formas: conhecimento, grana a mais e controle de tudo que vai acontecer na página.

**Comment by StanleySathler** [comment]
Já fiz alguns freelas. Algumas vezes, o designer quem mandava o layout. Outras, eu quem desenhava.

Nas vezes em que eu recebia o layout de um designer, esse problema acontecia bastante: ele não incluía as "variações". Desenhava um formulário de contato, mas não desenhava as mensagens de erro, não desenhava a aparência do formulário após o envio ser concluído com sucesso... e, naquela época, comecei a achar que era natural o desenvolvedor ter que se preocupar com esses detalhes.

Hoje, onde trabalho, o designer é responsável por tudo: ele precisa pensar nos diferentes estados da aplicação e desenhar cada um deles. Mensagens de erro, de sucesso... tudo. 

A única exceção é quando se trata de um componente... se uma determinada tela tem um dropdown, e o dropdown é aquele padrão que já temos na nossa lista de componentes, então o designer não precisa desenhar aquele dropdown aberto. Fora isso... tem que desenhar tudo.

Eu particularmente prefiro e acho correto. O designer é geralmente aquele que têm conhecimento de UX e que tá por dentro de todo o contexto visual do produto.

**Comment by felipefialho** [comment]
Pedir para abrir o Sketch ou Photoshop para criar um layout, nunca me pediram.

Mas é muito comum "desenhar" layouts diretamente no código, especialmente quando já tem todos os componentes visuais criados.

**Comment by phsantiago** [comment]
@raffaeldantass realmente, é bem melhor já ter o layout aprovado, modificações pelo código são muito mais custosas que modificações diretas no layout! Em um dos freelas que me pediram para fazer tudo, pela primeira vez comecei fazendo o layout no photoshop, para só partir para o código com a aprovação.

@kvnol Exato haha to vendo a vantagem de grana e conhecimento agora, eu quis começar pelo layout só para evitar modificações no código e por consequencia conheci essas outras vantagens.

@StanleySathler Que legal esse workflow de onde você trabalha! É difícil encontrar um lugar com designers que trabalham assim, até hoje não conheci nenhum. Você tem sorte rapaz hahah.

@LFeh comigo acontece o mesmo, nunca pediram para criar layout fora do código. Ultimamente tenho evitado ao máximo fazer direto no código, a chance de gerar retrabalho é muito grande e uma alteração por código é muito mais trabalhosa que uma no layout.

**Comment by raffaeldantass** [comment]
@phsantiago cara, é interessante haha 
Eu aprendi muita coisa desenvolvendo meus próprios layouts. Isso me ajudou a me aproximar do pessoal de design aqui do trabalho e agora consigo fazer comentários mais pontuais e dessa forma o trabalho fica muito mais fácil pra todo mundo. 
A gente consegue alinhar melhor Design/Código VS Prazo de entrega haha
Particularmente foi muito positivo pra mim haha

**Comment by didiraja** [comment]
Comigo sim, mas é porque eu era Designer antes de virar dev hahahah Inclusive quem não souber ou tiver dificuldade, só chamar ai pra gente desenrolar uns freelas 😉
